File: src/main/java/de/com/fdm/bot/CommandRunner.java
package de.com.fdm.bot;

import de.com.fdm.bot.commands.CommitCommand;
import de.com.fdm.bot.commands.EchoCommand;
import de.com.fdm.bot.commands.HttpStatusCommand;
import de.com.fdm.bot.commands.IdUserCommand;
import de.com.fdm.bot.commands.JoinChannelCommand;
import de.com.fdm.bot.commands.ListChannelsCommand;
import de.com.fdm.bot.commands.PartChannelCommand;
import de.com.fdm.bot.commands.PingCommand;
import de.com.fdm.bot.commands.UpdateCommand;
import de.com.fdm.bot.commands.UnkownCommand;
import de.com.fdm.bot.commands.UserIdCommand;
import de.com.fdm.twitch.tmi.TmiMessage;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import java.util.Arrays;
import java.util.HashMap;
import java.util.List;

@Component
public class CommandRunner {
    private final UnkownCommand unkownCommand;
    private final HashMap<String, Command> commandMap;

    public CommandRunner(
            @Autowired PingCommand pingCommand,
            @Autowired EchoCommand echoCommand,
            @Autowired UserIdCommand userIdCommand,
            @Autowired IdUserCommand idUserCommand,
            @Autowired HttpStatusCommand httpStatusCommand,
            @Autowired ListChannelsCommand listChannelsCommand,
            @Autowired JoinChannelCommand joinChannelCommand,
            @Autowired PartChannelCommand partChannelCommand,
            @Autowired CommitCommand commitCommand,
            @Autowired UpdateCommand updateCommand,
            @Autowired UnkownCommand unkownCommand
    ) {
        commandMap = new HashMap<>();
        commandMap.put("ping", pingCommand);
        commandMap.put("echo", echoCommand);
        commandMap.put("httpstatus", httpStatusCommand);
        commandMap.put("id", userIdCommand);
        commandMap.put("user", idUserCommand);
        commandMap.put("channels", listChannelsCommand);
        commandMap.put("joinchannel", joinChannelCommand);
        commandMap.put("partchannel", partChannelCommand);
        commandMap.put("commit", commitCommand);
        commandMap.put("update", updateCommand);

        this.unkownCommand = unkownCommand;
    }

    public String runCommand(TmiMessage msg) {
        String messageText = msg.getMessage().substring(1);
        String[] chunks = messageText.split(" ");

        List<String> args = Arrays.stream(chunks).toList().subList(1, chunks.length);
        Parameters params = new Parameters(msg.getChannel(), messageText, args);

        Command command = commandMap.getOrDefault(chunks[0], unkownCommand);

        return command.execute(params);
    }
}

File: src/main/java/de/com/fdm/bot/CommandRunner.kt
package de.com.fdm.bot

import de.com.fdm.bot.commands.*
import de.com.fdm.bot.commands.formula1.ConstructorStandingsCommand
import de.com.fdm.bot.commands.formula1.DriverStandingsCommand
import de.com.fdm.bot.api.twitch.tmi.TmiMessage
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component

@Component
class CommandRunner(
    @Autowired pingCommand: PingCommand,
    @Autowired echoCommand: EchoCommand,
    @Autowired userIdCommand: UserIdCommand,
    @Autowired idUserCommand: IdUserCommand,
    @Autowired httpStatusCommand: HttpStatusCommand,
    @Autowired listChannelsCommand: ListChannelsCommand,
    @Autowired joinChannelCommand: JoinChannelCommand,
    @Autowired partChannelCommand: PartChannelCommand,
    @Autowired commitCommand: CommitCommand,
    @Autowired updateCommand: UpdateCommand,
    @Autowired rawMsgCommand: RawMsgCommand,
    @Autowired driverStandingsCommand: DriverStandingsCommand,
    @Autowired constructorStandingsCommand: ConstructorStandingsCommand,
    @Autowired domainCommand: DynaDomainCommand,
    @Autowired unknownCommand: UnknownCommand
) {
    private val unknownCommand: UnknownCommand
    private val commandMap: HashMap<String, Command> = HashMap()

    init {
        commandMap["ping"] = pingCommand
        commandMap["echo"] = echoCommand
        commandMap["httpstatus"] = httpStatusCommand
        commandMap["id"] = userIdCommand
        commandMap["user"] = idUserCommand
        commandMap["channels"] = listChannelsCommand
        commandMap["joinchannel"] = joinChannelCommand
        commandMap["partchannel"] = partChannelCommand
        commandMap["commit"] = commitCommand
        commandMap["update"] = updateCommand
        commandMap["rawmsg"] = rawMsgCommand
        commandMap["driverstandings"] = driverStandingsCommand
        commandMap["constructorstandings"] = constructorStandingsCommand
        commandMap["domain"] = domainCommand

        this.unknownCommand = unknownCommand
    }

    fun runCommand(msg: TmiMessage): String {
        val messageText = msg.message.substring(1)
        val chunks = messageText.split(" ")
        val command = commandMap.getOrDefault(chunks[0], unknownCommand)
        return command.execute(msg)
    }
}